ddl/index.go

```diff
@@ -49,6 +49,10 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.Gen("column does not exist: %s", ic.Column.Name)
 		}
 
+		if col.Flen == 0 {
```

Comment:
You can check the Flen in this function:https://github.com/pingcap/tidb/blob/master/plan/validator.go#L259

Refer to [this](https://github.com/pingcap/tidb/blob/master/plan/validator.go#L217) as an example.